File: Mikasa's Race to Coast/claude2.cpp
#include <bits/stdc++.h>
using namespace std;
typedef long long ll;

int main() {
    ios_base::sync_with_stdio(false);
    cin.tie(NULL);
    
    int T;
    cin >> T;
    
    while (T--) {
        int N;
        ll D, F;
        cin >> N >> D >> F;
        
        vector<pair<ll, ll>> depots(N);
        for (int i = 0; i < N; i++) {
            cin >> depots[i].first >> depots[i].second;
        }
        
        sort(depots.begin(), depots.end());
        
        priority_queue<ll> maxHeap;
        ll currentFuel = F;
        ll currentPosition = 0;
        int stopCount = 0;
        bool possible = true;
        
        for (int i = 0; i <= N; i++) {
            ll targetPosition = (i == N) ? D : depots[i].first;
            ll distanceNeeded = targetPosition - currentPosition;
            
            while (currentFuel < distanceNeeded) {
                if (maxHeap.empty()) {
                    possible = false;
                    break;
                }
                currentFuel += maxHeap.top();
                maxHeap.pop();
                stopCount++;
            }
            
            if (!possible) {
                break;
            }
            
            currentFuel -= distanceNeeded;
            currentPosition = targetPosition;
            
            if (i < N) {
                maxHeap.push(depots[i].second);
            }
        }
        
        cout << (possible ? stopCount : -1) << "\n";
    }
    
    return 0;
}

File: Mikasa's Race to Coast/claude2.java
import java.util.*;
import java.io.*;

public class Main {
    public static void main(String[] args) throws IOException {
        BufferedReader reader = new BufferedReader(new InputStreamReader(System.in));
        StreamTokenizer tokenizer = new StreamTokenizer(reader);
        StringBuilder output = new StringBuilder();
        
        tokenizer.nextToken();
        int T = (int) tokenizer.nval;
        
        for (int testCase = 0; testCase < T; testCase++) {
            tokenizer.nextToken();
            int N = (int) tokenizer.nval;
            tokenizer.nextToken();
            long D = (long) tokenizer.nval;
            tokenizer.nextToken();
            long F = (long) tokenizer.nval;
            
            long[][] depots = new long[N][2];
            for (int i = 0; i < N; i++) {
                tokenizer.nextToken();
                depots[i][0] = (long) tokenizer.nval;
                tokenizer.nextToken();
                depots[i][1] = (long) tokenizer.nval;
            }
            
            Arrays.sort(depots, (a, b) -> Long.compare(a[0], b[0]));
            
            PriorityQueue<Long> maxHeap = new PriorityQueue<>(Collections.reverseOrder());
            long currentFuel = F;
            long currentPosition = 0L;
            int stopCount = 0;
            boolean possible = true;
            
            for (int i = 0; i <= N; i++) {
                long targetPosition = (i == N) ? D : depots[i][0];
                long distanceNeeded = targetPosition - currentPosition;
                
                while (currentFuel < distanceNeeded) {
                    if (maxHeap.isEmpty()) {
                        possible = false;
                        break;
                    }
                    currentFuel += maxHeap.poll();
                    stopCount++;
                }
                
                if (!possible) {
                    break;
                }
                
                currentFuel -= distanceNeeded;
                currentPosition = targetPosition;
                
                if (i < N) {
                    maxHeap.add(depots[i][1]);
                }
            }
            
            output.append(possible ? stopCount : -1).append("\n");
        }
        
        System.out.print(output);
    }
}